src/test/java/com/mariamkatamashvlii/gym/controller/TraineeControllerTest.java

```diff
@@ -0,0 +1,144 @@
+package com.mariamkatamashvlii.gym.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mariamkatamashvlii.gym.dto.RegistrationResponseDTO;
+import com.mariamkatamashvlii.gym.dto.ToggleActivationDTO;
+import com.mariamkatamashvlii.gym.dto.traineeDto.ProfileResponseDTO;
+import com.mariamkatamashvlii.gym.dto.traineeDto.RegistrationRequestDTO;
+import com.mariamkatamashvlii.gym.service.TraineeService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.aot.DisabledInAotMode;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.sql.Date;
+import java.util.ArrayList;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration(classes = {TraineeController.class})
+@ExtendWith(SpringExtension.class)
+@DisabledInAotMode
+class TraineeControllerTest {
+    @Autowired
+    private TraineeController traineeController;
+
+    @MockBean
+    private TraineeService traineeService;
+
+    @Test
+    void testDelete() throws Exception {
+        doNothing().when(traineeService).delete(Mockito.<String>any());
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.delete("/traineeProfile/{username}/delete",
+                "janedoe");
+
+        MockMvcBuilders.standaloneSetup(traineeController)
+                .build()
+                .perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().contentType("text/plain;charset=ISO-8859-1"))
+                .andExpect(MockMvcResultMatchers.content().string("Trainee deleted successfully"));
+    }
+
+    @Test
+    void testDelete2() throws Exception {
```

Comment:
this name recalls me famous anekdote about son1 and son2.

Tests names should tell about the scenario. F.e.
testDelete_whenTraneeDoesNotExist_thenExceptioThrown